### DISCUSSION TITLE
No new (auto) version for Cask

### DISCUSSION THREAD

**Discussion by joh-klein** [Homebrew/discussions#2033] [Casks]
I added a cask in November 2020 and it got updated automatically for a while. Today I saw, that a couple of newer versions were not picked up. I thought that happend now automatically?

The cask in question is [sky ticket](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/sky-ticket.rb)

It seems like the auto update functionality was changed. Where can I find out, that these kinds of changes happen?

**Comment by SMillerDev** [comment]
I don't think casks ever automatically updated. I guess someone just updated it for you for a while. You can run `brew livecheck <cask>` to see if brew can automatically find new versions, that should make it easier for people to contribute.

**Reply by joh-klein** [reply]
I thought the "BrewTestBot" did a daily URL check or something.

**Reply by SMillerDev** [reply]
Seeing the history it might have done that for a while, but since there is no way to identify updates for Sky Ticket it's not possible to update it (automatically) anymore.

**Reply by joh-klein** [reply]
And that is the second part of my question: where can I find out, that this changed? Or was that never something official?

**Reply by SMillerDev** [reply]
If you're refering to the testbot auto-updates changing, that was likely always on a best-effort basis and after https://brew.sh/2021/04/21/security-incident-disclosure/ we disabled a bunch of automation for casks.

As for your specific case, upstream changed their download URL and nobody found a way to check the version anymore. You'd have to ask them when/why that changed.

**Reply by joh-klein** [reply]
You mean the URL in the cask file, right? That is still a valid and working URL …

**Reply by SMillerDev** [reply]
Yes, but it doesn't have a version like before. So how is brew supposed to know what version is the latest version?

**Reply by SMillerDev** [reply]
Actually, it never did. I have no idea how it came up with a version, or with updates then.

**Reply by joh-klein** [reply]
OK, I found, that they also have a versioned URL:
https://web.static.nowtv.com/watch/player/skyticket/de/8.6.0/SkyTicket-Mac.dmg

How do I right a livecheck for that?

**Reply by SMillerDev** [reply]
Where did you find the versioned URL? Because that would be the place that livecheck needs to check.

And the cask download url would need to change to that one.

**Reply by joh-klein** [reply]
When the app starts it GETs this file

https://web.static.nowtv.com/watch/player/skyticket/de/latest/update.json

and in there are the latest versions

**Reply by SMillerDev** [reply]
You can do something like this cask: https://github.com/Homebrew/homebrew-cask/blob/03bd261c8650eebd01791adae9d86013cd4506d8/Casks/hbuilderx.rb

**Reply by joh-klein** [reply]
Great, thanks! How can I test drive the json parsing?

**Reply by SMillerDev** [reply]
`brew livecheck --debug <name>`

**Reply by joh-klein** [reply]
Thanks for all your help. I added a PR with the necessary changes.

https://github.com/Homebrew/homebrew-cask/pull/110563
